src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests.cs

```diff
@@ -12174,5 +12174,50 @@ public static void Main(string[] args)
                 );
             var comp = CompileAndVerify(compilation, expectedOutput: "whatever");
         }
+
+        [Fact, WorkItem(12996, "https://github.com/dotnet/roslyn/issues/12996")]
+        public void TypeOfAVarPatternVariable()
```

Comment:
>TypeOfAVarPatternVariable [](start = 20, length = 25)

I would expect to see tests for more scenarios. At the very least, please clone tests Scope_SwitchLabelGuard_02 and the rest of them with "Scope_SwitchLabelGuard_" prefix.